src/unix.rs

```diff
@@ -61,3 +61,27 @@ pub fn try_wait_without_reaping(handle: Handle) -> io::Result<bool> {
                            format!("unexpected si_signo from waitid: {}", siginfo.si_signo)))
     }
 }
+
+pub fn terminate(child: &super::SharedChild) -> io::Result<()> {
+    child.send_signal(libc::SIGTERM)
+}
+
+pub trait SharedChildExt {
+    /// Sends a signal to the child process with `libc::kill`.
+    fn send_signal(&self, signal: libc::c_int) -> io::Result<()>;
+}
+
+impl SharedChildExt for super::SharedChild {
+    fn send_signal(&self, signal: libc::c_int) -> io::Result<()> {
+        let status = self.state_lock.lock().unwrap();
+        if let super::ChildState::Exited(_) = *status {
+            return Ok(());
```

Comment:
I would expect it to work the same way as the default `Child`. And it feels strange that it returns `Ok` after not even sending the signal. But on the other hand it's easier to write code for concurrent wait and signal/kill if it returns `Ok`. Going with the current `Ok` approach makes it impossible to know if the signal was actually sent or not. I don't know if that is a drawback though. For my current use case I don't care if it was sent or not, I just want to tell it to quit gracefully if possible.